### PR TITLE
Website: Fix bug in usersearch

### PR DESCRIPTION
**Pull request by AnnikaCodes**
`$ip` isn't defined here - the correct behavior is to search by the userlist we just fetched.

### CODE REVIEW

File: website/usersearch.php [hunk 1]
```diff
@@ -171,7 +171,10 @@
 	$userlist = $psdb->query("SELECT `ntbb_users`.`username`, `ntbb_users`.`userid`, `ntbb_users`.`banstate` FROM `ntbb_users` INNER JOIN `ntbb_usermodlog` ON `ntbb_users`.`userid` = `ntbb_usermodlog`.`userid` WHERE `entry` LIKE ?", [$entry]);
 	if ($csrfOk && isset($_POST['standing'])) {
 		$newStanding = intval($_POST['standing']);
-		$psdb->query("UPDATE ntbb_users SET banstate = ? WHERE ip = ? AND banstate != 100", [$newStanding, $ip]);
+		$psdb->query(
+			"UPDATE ntbb_users SET banstate = ? WHERE userid IN (" . implode(',', $userlist) . ") AND banstate != 100",
```
Comment (Zarel) [inline website/usersearch.php:175]: Err, can `$userlist` be used this way? Isn't it a database resource, not an array? I also don't think it's properly escaped...

Comment (AnnikaCodes) [reply website/usersearch.php:175]: Oh, yeah, I meant to convert it to an array; I'll do that now.

File: website/usersearch.php [hunk 2]
```diff
@@ -171,16 +171,22 @@
 	$userlist = $psdb->query("SELECT `ntbb_users`.`username`, `ntbb_users`.`userid`, `ntbb_users`.`banstate` FROM `ntbb_users` INNER JOIN `ntbb_usermodlog` ON `ntbb_users`.`userid` = `ntbb_usermodlog`.`userid` WHERE `entry` LIKE ?", [$entry]);
 	if ($csrfOk && isset($_POST['standing'])) {
 		$newStanding = intval($_POST['standing']);
-		$psdb->query("UPDATE ntbb_users SET banstate = ? WHERE ip = ? AND banstate != 100", [$newStanding, $ip]);
 
+		$users = array();
```
Comment (Zarel) [inline website/usersearch.php:175]:
```suggestion
		$users = [];
```

File: website/usersearch.php [hunk 3]
```diff
@@ -171,16 +171,22 @@
...
 		foreach ($userlist as $row) {
 			$modlogentry = "Standing changed to $newStanding ({$STANDINGS[$newStanding]}): {$_POST['allreason']}";
 			$psdb->query(
 				"INSERT INTO `{$psdb->prefix}usermodlog` (`userid`,`actorid`,`date`,`ip`,`entry`) VALUES (?, ?, ?, ?, ?)",
 				[$row['userid'], $curuser['userid'], time(), $users->getIp(), $modlogentry]
 			);
+			array_push($users, $psdb->escape($row['userid']));
```
Comment (Zarel) [inline website/usersearch.php:182]:
```suggestion
			$users []= $psdb->escape($row['userid']);
```

**Comment by Zarel** [comment]
Is there any reason why we can't define `$ip` here? I'm not a big fan of manually constructing an unbounded string to be executed from an array, even if there are no longer any security issues.

**Comment by Zarel** [comment]
(Sorry, last time I reviewed I was still confused about exactly what this was doing any why; I understand what this is doing now.)

**Comment by AnnikaCodes** [comment]
> Is there any reason why we can't define `$ip` here? I'm not a big fan of manually constructing an unbounded string to be executed from an array, even if there are no longer any security issues.

Why would we define `$ip`? As far as I know, this code is searching by modlog note, not by IP address?

**Comment by Zarel** [comment]
This code is designed to change the standing of all users with the same registration IP at once; to permalock a user with a lot of alts.

**Comment by Zarel** [comment]
(It's also been working fine this entire time, so I'm also still confused about what bug you're seeing.)

**Comment by AnnikaCodes** [comment]
I haven't seen a bug personally - it was reported by pre and @asgdf.
image

**Comment by Zarel** [comment]
Oh wait I did get it mixed up with the one I wrote.

I wonder if you should just write a new query for that one.

**Comment by Zarel** [comment]
I think you can do this with a JOIN.

**Comment by AnnikaCodes** [comment]
Alright.

**Comment by Zarel** [comment]
Thanks! I like this way a lot better.

**Pull request merged**